pytorch_lightning/trainer/data_loading.py

```diff
@@ -163,10 +163,22 @@ def reset_val_dataloader(self, model):
         # determine number of validation batches
         # val datasets could be none, 1 or 2+
         if self.val_dataloaders is not None:
-            self._percent_range_check('val_percent_check')
+            for dataloader in self.val_dataloaders:
+                if self.is_infinite_dataloader(dataloader):
+                    self.num_val_batches = float('inf')
```

Comment:
```suggestion
                    self.num_val_batches = float('inf')
                    break
```

Reply:
Done :)

Reply:
btw, you can also accept suggestion here at once lol